openapi/nexus.json

```diff
@@ -15970,7 +15970,6 @@
           "signing_keypair": {
             "nullable": true,
             "description": "request signing key pair",
-            "default": null,
```

Comment:
notable?

Reply:
Unsure! Let me see if I can find out why this changed.

Reply:
First guess would be the schemars update.

Reply:
Yeah, it seems to have been the one PR in 0.18.8, which caused other breakage: https://github.com/oxidecomputer/dropshot/issues/995

I don't know whether removing `"default": null` is problematic on its own.

Reply:
The Progenitor-generated `oxide-client` (that we have in this repo to use for end-to-end tests) doesn't have any differences from main to this branch.

The generated API in the console does have this difference:

```
diff --git a/app/api/__generated__/validate.ts b/app/api/__generated__/validate.ts
index fcb58fb1..07fe33d0 100644
--- a/app/api/__generated__/validate.ts
+++ b/app/api/__generated__/validate.ts
@@ -2327,7 +2327,7 @@ export const SamlIdentityProviderCreate = z.preprocess(
     idpEntityId: z.string(),
     idpMetadataSource: IdpMetadataSource,
     name: Name,
-    signingKeypair: DerEncodedKeyPair.default(null).optional(),
+    signingKeypair: DerEncodedKeyPair.optional(),
     sloUrl: z.string(),
     spClientId: z.string(),
     technicalContactEmail: z.string(),
```

which seems to not introduce any typechecking issues.

Reply:
You would think the default value of a nullable would be null by default anyway! Thanks for checking the TS. That particular bit isn’t load-bearing anyway — we currently only use the generated validators in the mock server so we can 400 on bad request bodies.